merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/CachedSimulationService.java

```diff
@@ -33,11 +27,11 @@ public ResultsProtocol.State getSimulationResults(final PlanId planId, final Rev
   }
 
   @Override
-  public <T> Optional<T> get(final PlanId planId, final RevisionData revisionData, final Function<SimulationResults, T> getter) {
+  public Optional<SimulationResults> get(final PlanId planId, final RevisionData revisionData) {
     return this.store.lookup(planId) // Only return results that have already been cached
         .map(ResultsProtocol.ReaderRole::get)
         .map(state -> state instanceof final ResultsProtocol.State.Success s ?
-            getter.apply(s.results()) :
+            s.results() :
```

Comment:
Why did you make this change?

Reply:
To sum up my conversation above, I moved the getter callback outside of this `get()` function, since to get the full results object out of this getter would require a somewhat awkward
```java
final var results = simulationService
    .get(planId, revisionData, Function.identity())
    .orElseGet(Collections::emptyMap);
```
so now, instead of calling
```java
final var simActs = simulationService
    .get(planId, revisionData, r -> r.simulatedActivities)
    .orElseGet(Collections::emptyMap);
```
you can use
```java
final var simActs = simulationService
    .get(planId, revisionData)
    .map(r -> r.simulatedActivities)
    .orElseGet(Collections::emptyMap);
```
which allows easy extraction of the full `SimulationResults` object if needed.